modules/agent/pkg/controllers/bundledeployment/controller.go

```diff
@@ -252,3 +275,111 @@ func readyError(status fleet.BundleDeploymentStatus) error {
 
 	return errors.New(msg)
 }
+
+func hasAutoReapply(bd *fleet.BundleDeployment, status *fleet.BundleDeploymentStatus) bool {
+	if bd.Spec.Options.ResyncPolicy == nil && bd.Spec.StagedOptions.ResyncPolicy == nil {
+		if status.LastApply == nil {
+			return false
+		}
+		if status.LastApply.Add(DefaultReapplyInterval).Before(time.Now()) {
+			return true
+		}
+		// use custom resync policy from the template //
+		policy := &fleet.ResyncPolicy{}
+		if bd.Spec.Options.ResyncPolicy != nil {
+			policy = bd.Spec.Options.ResyncPolicy
+		}
+
+		if bd.Spec.StagedOptions.ResyncPolicy != nil {
+			policy = bd.Spec.StagedOptions.ResyncPolicy
+		}
+
+		if policy.MaxRetries == 0 {
+			return false
+		}
+
+		if policy.MaxRetries == DefaultMaxRetries && status.LastApply.Add(policy.ResyncDelay.Duration).Before(time.Now()) {
```

Comment:
Hmm why do we even compare `policy.MaxRetries == DefaultMaxRetries` here?

Reply:
DefaultMaxRetries was a specific case where you specify MaxRetries as `-1` which means its an infinite loop where the objects will be queued at the custom reapply interval.